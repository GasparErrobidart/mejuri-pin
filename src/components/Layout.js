import React       from 'react';
import Container   from './Container';
import Header      from './Header';

const Layout = ({children}) =>{
  return(
    <React.Fragment>
      <div className="app-wrapper">
        <Header />
        <main className="main-content">
          {children}
        </main>
        <footer className="main-footer">
          <Container>
          </Container>
        </footer>
      </div>
    </React.Fragment>
  )
}

export default Layout;
