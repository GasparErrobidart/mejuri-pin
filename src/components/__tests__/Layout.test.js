import '@testing-library/jest-dom'
import 'jest-styled-components'

import configureStore    from 'redux-mock-store'
import React             from 'react'
import { cleanup }       from '@testing-library/react'
import { Provider }      from 'react-redux'
import renderer          from 'react-test-renderer'
import {ThemeProvider}   from 'styled-components'

import Layout            from '../Layout'
import theme             from '../GlobalStyle/themes/mejuri-pin'

const mockStore = configureStore([]);


describe('Layout component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      likes: [],
    });
  });

  it("Should render",()=>{
    const testText        = "Layout Test Text"
    const component       = renderer.create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Layout>{testText}</Layout>
        </ThemeProvider>
      </Provider>
    );
    const tree            = component.toJSON()
    expect(tree).toMatchSnapshot();
  })

  afterEach(cleanup)


});
