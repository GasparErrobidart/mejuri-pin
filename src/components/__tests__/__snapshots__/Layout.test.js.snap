// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Should render 1`] = `
.c0 {
  padding: 0 20px;
  box-sizing: border-box;
  width: 100%;
}

.c3 {
  height: 50px;
  padding: 5px 0;
}

.c3 > img {
  height: 100%;
  width: auto;
}

.c2 {
  font-family: 'Lato',sans-serif;
  font-size: 30px;
  text-align: left;
  position: relative;
  cursor: pointer;
}

.c2 *[animated-heart] {
  color: red;
  position: absolute;
  opacity: 0;
}

.c2 *[animated-heart].animating {
  -webkit-animation: pop 700ms;
  animation: pop 700ms;
}

.c2 > span {
  position: absolute;
  font-size: 12px;
  display: inline-block;
  padding: 2px;
  font-weight: 400;
  margin-left: 22px;
  margin-top: 10px;
  text-align: center;
  border-radius: 2px;
  background: #888888;
  color: white;
  left: 0;
  top: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 > * {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > *:nth-child(odd) {
  -webkit-flex: 0 0 55px;
  -ms-flex: 0 0 55px;
  flex: 0 0 55px;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c1 > *:nth-child(2) {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

<div
  className="app-wrapper"
>
  <header
    className="main-header"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        <div
          className="c2"
          onClick={[Function]}
        >
          <div
            animated-heart="true"
            className=""
            onAnimationEnd={[Function]}
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
              />
            </svg>
          </div>
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
            />
          </svg>
          <span>
            0
          </span>
        </div>
        <div>
          <div>
            <div
              className="c3"
            >
              <img
                alt="Mejuri Logo"
                src="/mejuri.svg"
              />
            </div>
          </div>
        </div>
        <div />
      </div>
    </div>
  </header>
  <main
    className="main-content"
  >
    Layout Test Text
  </main>
  <footer
    className="main-footer"
  >
    <div
      className="c0"
    />
  </footer>
</div>
`;
